.github/workflows/build_workflow.yml

```diff
@@ -0,0 +1,305 @@
+        if: always()
+        with:
+          name: 'Checkstyle Reports'
+          path: "./service/build/reports/**"
+          compression-level: 9
+
+  spotbugs:
+    name: Spotbugs
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+
+      - name: Setup Java 21 LTS
+        uses: actions/setup-java@v4
+        with:
+          java-version: 21
+          distribution: 'corretto'
+
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v4
+
+      - name: Spotbugs
+        run: |
+          ./gradlew spotbugsMain
+          ./gradlew spotbugsTest
+        working-directory: ./service
+
+      - name: Upload Artifacts
+        uses: actions/upload-artifact@v4
+        if: always()
+        with:
+          name: 'Spotbugs Reports'
+          path: "./service/build/reports/**"
+          compression-level: 9
+
+# ------------------------------------------------------------
+
+  unit-tests:
+    name: Unit Tests
+    runs-on: ubuntu-latest
+    needs: [checkstyle, spotbugs]
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+
+      - name: Setup Java 21 LTS
+        uses: actions/setup-java@v4
+        with:
+          java-version: 21
+          distribution: 'corretto'
+
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v4
+
+      - name: Execute Unit Tests
+        run: ./gradlew test
+        working-directory: ./service
+
+      - name: Upload Artifacts
+        uses: actions/upload-artifact@v4
+        if: always()
+        with:
+          name: 'Unit Test Reports'
+          path: |
+            ./service/build/reports/**
+            ./scripts/logs/**
+          compression-level: 9
+
+  integration-tests:
+    name: Integration Tests
+    runs-on: ubuntu-latest
+    needs: [checkstyle, spotbugs]
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+
+      - name: Setup Java 21 LTS
+        uses: actions/setup-java@v4
+        with:
+          java-version: 21
+          distribution: 'corretto'
+
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v4
+
+      - name: Start Docker Dependencies
+        env:
+          GPC_SUPPLIER_ODS_CODE: "XYZ222"
+          GP2GP_SERVER_PORT: "8080"
+          GP2GP_AMQP_BROKERS: "amqp://activemq:5672"
+          GP2GP_MONGO_URI: "mongodb://mongodb:27017"
+          GP2GP_MONGO_DATABASE_NAME: "gp2gp"
+          GP2GP_MHS_OUTBOUND_URL: "http://mock-mhs-adaptor:8081/mock-mhs-endpoint/accept-only"
+          GP2GP_GPC_GET_URL: "http://gpcc:8090/@ODS_CODE@/STU3/1/gpconnect"
+          GP2GP_LARGE_ATTACHMENT_THRESHOLD: "4500000"
+          GP2GP_LARGE_EHR_EXTRACT_THRESHOLD: "4500000"
+          GPC_CONSUMER_SERVER_PORT: "8090"
+          GPC_CONSUMER_SDS_URL: "http://wiremock:8080/spine-directory/"
+          GPC_CONSUMER_SDS_APIKEY: "anykey"
+          GP2GP_LOGGING_LEVEL: DEBUG
+          GPC_CONSUMER_LOGGING_LEVEL: DEBUG
```

Comment:
Investigate if there is an alternative way to get these environment variables imported via a file on the LFS - without using `source`.

Reply:
Hmm, I'm a little suspicious of the suggestion of using LFS. Not sure how it would apply here.

Reply:
I think the variable can get imported using ".env" file.

Reply:
Also, we can make logging levels adjustable:
`GP2GP_LOGGING_LEVEL: ${{ secrets.GP2GP_LOGGING_LEVEL || 'DEBUG' }}
 GPC_CONSUMER_LOGGING_LEVEL: ${{ secrets.GPC_CONSUMER_LOGGING_LEVEL || 'DEBUG' }}`